xla/service/gpu/gpu_executable.cc

```diff
@@ -825,7 +829,10 @@ Status GpuExecutable::SetUpMlirAllocation(
     absl::flat_hash_map<ShapeIndex, GpuExecutable::OutputInfo>* output_info,
     Shape* output_shape) {
   for (int i = 0; i < buffer_sizes.size(); i++) {
-    allocations->emplace_back(i, buffer_sizes[i], 0);
+    // This code path is taken when using the non-thunk based runtime. Memory
+    // space is being set to 0 for all allocations. We need to copy over the
+    // value from BufferAssignment instead.
+    allocations->emplace_back(i, buffer_sizes[i], /*memory_space=*/0);
```

Comment:
Could you clarify? This feature is only working for the thunk runtime?

Reply:
Yes that's correct.

The `colorerer` for BufferAssignments sets the memory space values for the allocations.
For the thunk runtime, those allocations are directly copied over to the executable and the memory space values are preserved.
However, for the non-thunk runtime, the allocations are reconstructed during `GetMlirAllocationInfo` and the memory space value gets erroneously set to 0 at this line.